zombies.js

```diff
@@ -1,3 +1,162 @@
+// jshint esversion: 6
+class Item {
+  constructor(name) {
+    this.name = name;
+  }
+}
+
+class Weapon extends Item {
+  constructor(name, damage) {
+    super(name);
+    this.damage = damage;
+  }
+}
+
+class Food extends Item {  
+  constructor(name, energy) {
+    super(name);
+    this.energy = energy;
+  }
+}
+
+class Player {
+  constructor(name, health, strength, speed) {
+    this._pack = [];
+    this._maxHealth = health;
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this.equipped = false;
+  }
+
+  getPack() {
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  checkPack() {
+    let playerPack = this._pack;
+    console.log('Contents: ');
+    for (let i in playerPack.length) {
+      console.log(playerPack[i] + ", ");
+    }
+  }
+
+  takeItem(item) {
+    let packContents = this._pack;
+    if (packContents.length > 2 && !packContents.includes(item)) {
+      console.log('Your pack is full.');
+      return false;
+    } else {
+      packContents.push(item);
+      console.log(this.name + ' picked up ' + item.name);
+      return true;
+    }
+  }
+
+  discardItem(item) {
+    let itemIndex = this._pack.indexOf(item);
+
+    if (itemIndex !== -1) {
+      this._pack.splice(itemIndex, 1);
+      console.log(this.name + ' ' + item.name + ' was discarded.');
+      return true;
+    } else {
+      console.log(item.name + ' nothing was discarded since the item could not be found.');
+      return false;
+    }
+  }
+
+  equip(itemToEquip) {
+    let itemIndex = this._pack.indexOf(itemToEquip);
+    let equipped = this.equipped;
+    let _pack = this._pack;
+
+    if (!equipped && _pack.includes(itemToEquip) && itemToEquip instanceof Weapon) {
```

Comment:
`_pack.includes(itemToEquip` this test could be replace with `itemIndex > -1` since you already check the pack for an item.